test/functional/lua/diagnostic_spec.lua

```diff
@@ -1654,6 +1667,55 @@ describe('vim.diagnostic', function()
   end)
 
   describe('config()', function()
+    describe('underline severity_sort', function()
+      before_each(function()
+        exec_lua(function()
+          vim.diagnostic.config({
+            underline = true,
+            severity_sort = true,
+          })
+
+          vim.diagnostic.set(_G.diagnostic_ns, _G.diagnostic_bufnr, {
+            _G.make_warning('Some Warning', 0, 0, 0, 0),
+            _G.make_error('Some Error', 0, 0, 0, 0),
+            _G.make_hint('Some Hint', 0, 0, 0, 0),
+          })
+        end)
+      end)
+
+      it('underline respects severity_sort reverse=false', function()
+        eq(
+          'DiagnosticUnderlineError',
+          exec_lua(function()
+            local extmarks = _G.get_underline_extmarks(_G.diagnostic_ns)
+            table.sort(extmarks, function(a, b)
+              return a[4].priority > b[4].priority
+            end)
+            return extmarks[1][4].hl_group
+          end)
+        )
+      end)
+
+      it('underline respects severity_sort reverse=true', function()
+        exec_lua(function()
+          vim.diagnostic.config({
+            severity_sort = { reverse = true },
+          })
+        end)
+
+        eq(
+          'DiagnosticUnderlineHint',
+          exec_lua(function()
+            local extmarks = _G.get_underline_extmarks(_G.diagnostic_ns)
+            table.sort(extmarks, function(a, b)
+              return a[4].priority > b[4].priority
+            end)
+            return extmarks[1][4].hl_group
+          end)
+        )
+      end)
+    end)
+
```

Comment:
Can you move this to the "allows sorting by severity" section?

Reply:
will do